packages/web-services-mongodb-remote/src/lib/internal/RemoteMongoCollectionImpl.ts

```diff
@@ -0,0 +1,160 @@
+import { Codec } from "stitch-core";
+import {
+  CoreRemoteMongoCollection,
+  RemoteCountOptions,
+  RemoteDeleteResult,
+  RemoteFindOptions,
+  RemoteInsertManyResult,
+  RemoteInsertOneResult,
+  RemoteUpdateOptions,
+  RemoteUpdateResult
+} from "stitch-core-services-mongodb-remote";
+import RemoteMongoCollection from "../RemoteMongoCollection";
+import RemoteMongoReadOperation from "../RemoteMongoReadOperation";
+
+export default class RemoteMongoCollectionImpl<DocumentT> {
+  /**
+   * Gets the namespace of this collection.
+   *
+   * @return the namespace
+   */
+  public readonly namespace: string = this.proxy.namespace;
+
+  constructor(
+    private readonly proxy: CoreRemoteMongoCollection<DocumentT>,
+    private readonly codec?: Codec<DocumentT>
+  ) {}
+  /**
+   * Create a new CoreRemoteMongoCollection instance with a different default class to cast any
+   * documents returned from the database into.
+   *
+   * @param codec the default class to decode any documents returned from the database into.
+   * @param <NewDocumentT> the type that the new collection will encode documents from and decode
+   *                      documents to.
+   * @return a new CoreRemoteMongoCollection instance with the different default class
+   */
+  public withCollectionType<U>(codec: Codec<U>): RemoteMongoCollection<U> {
+    return new RemoteMongoCollectionImpl(
+      this.proxy.withCollectionType(codec),
+      codec
+    );
+  }
+
+  /**
+   * Counts the number of documents in the collection.
+   * @param query the query filter
+   * @param options the options describing the count
+   *
+   * @return the number of documents in the collection
+   */
+  public count(query?: object, options?: RemoteCountOptions): Promise<number> {
+    return this.proxy.count(query, options);
+  }
+
+  /**
+   * Finds all documents in the collection.
+   *
+   * @param query the query filter
+   * @return the find iterable interface
+   */
+  public find(
+    query?: object,
```

Comment:
[nit] In Java and Swift we've been calling this `filter`. I think we should do the same here.